web/__init__.py

```diff
@@ -72,36 +136,93 @@ def ctrl(sock):
 
 @app.get("/video")
 def video_download():
-
+    """
+    Handles the video streaming/downloading feature in the Flask app
+    """
     def generate():
+        if not app.config["login"]:
+            return
         for msg in app.svc.stream("videoqueue"):
             yield msg.data
 
-    return Response(generate(), mimetype='video/mp4')
+    return Response(generate(), mimetype="video/mp4")
 
 
 @app.get("/")
 def app_root():
-    host = request.host.split(':')
-    requestPort = host[1] if len(host) > 1 else '80' # If there is no 2nd array entry, the request port is 80
-    return render_template(
-        "index.html",
-        requestPort=requestPort,
-        requestHost=host[0]
-    )
+    """
+    Renders the html template for the root route, which is the homepage of the Flask app
+    """
+    config = app.config["config"]
+    with config.open() as cfg:
+        user_agent = user_agent_parse(request.headers.get("User-Agent"))
+        user_os = web.platform.os_platform(user_agent.os.family)
+
+        host = request.host.split(":")
+        # If there is no 2nd array entry, the request port is 80
+        request_port = host[1] if len(host) > 1 else "80"
+        if app.config["login"]:
+            anker_config = str(web.config.config_show(cfg))
+            printer_serial = str(cfg.printers[0].sn)
+        else:
+            anker_config = "<p>No printers found, please load your login config...</p>"
+            printer_serial = None
+
+        return render_template(
+            "index.html",
+            request_port=request_port,
+            request_host=host[0],
+            configure=app.config["login"],
+            login_file_path=web.platform.login_path(user_os),
+            anker_config=anker_config,
+            printer_serial=printer_serial
+        )
 
 
 @app.get("/api/version")
 def app_api_version():
-    return {
-        "api": "0.1",
-        "server": "1.9.0",
-        "text": "OctoPrint 1.9.0"
-    }
+    """
+    Returns the version details of api and server as dictionary
+
+    Returns:
+        A dictionary containing version details of api and server
+    """
+    return {"api": "0.1", "server": "1.9.0", "text": "OctoPrint 1.9.0"}
+
+
+@app.post("/api/web/config/upload")
```

Comment:
As I mentioned, we should have `/ankerctl/` in the api scope for our internal endpoints. For example, `/api/ankerctl/config/upload` for this endpoint.

Since `/api` is used for compatibility with OctoPrint, we have to be quite sure we don't have a name clash with OctoPrint, and `/web` is certainly generic enough that it might be used.

We could also move our scope somewhere else, since it's entirely under our control, but it would make sense to keep it under `/api`, since that might very well be the authenticated scope if we later add a login system.

So please change to `/api/ankerctl/config/upload`, or other reasonable non-clashing name.

(same kind of change needed for other ankerctl-specific endpoints)

Reply:
Do you also want `/reload` to become `/ankerctl/reload`?

Reply:
Yes - and I can see it's already done. Great :+1: